sql/parser2/encode.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2012, Google Inc. All rights reserved.
```

Comment:
I really should be including both copyright notices everywhere. This code originally came from vitess, so did some of the other syntax tree code, though there are starting to be modifications.